File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreDataWordVB/ThisDocument.vb
Public Class ThisDocument

    Private Sub ThisDocument_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
        CheckSchema()
    End Sub

    '--------------------------------------------------------------------------
    '<Snippet1>
    ' Ensure that the schema is in the library and registered with the document.
    Private Function CheckSchema() As Boolean

        Const namespaceUri As String = "http://schemas.contoso.com/projects"
        Dim namespaceFound As Boolean = False
        Dim namespaceRegistered As Boolean = False

        Dim n As Word.XMLNamespace
        For Each n In Application.XMLNamespaces
            If (n.URI = namespaceUri) Then
                namespaceFound = True
            End If
        Next

        If Not namespaceFound Then
            MessageBox.Show("XML Schema is not in library.")
            Return False
        End If

        Dim r As Word.XMLSchemaReference
        For Each r In Me.XMLSchemaReferences
            If (r.NamespaceURI = namespaceUri) Then
                namespaceRegistered = True
            End If
        Next

        If Not namespaceRegistered Then
            MessageBox.Show("XML Schema is not registered for this document.")
            Return False
        End If

        Return True
    End Function
    '</Snippet1>

End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreDataWordCS/ThisDocument.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word=Microsoft.Office.Interop.Word;
using Office=Microsoft.Office.Core;

namespace Trin_VstcoreDataWordCS
{
    public partial class ThisDocument
    {
        private void InternalStartup()
        {
            this.Startup+=new System.EventHandler(ThisDocument_Startup);
        }

        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            CheckSchema();
        }

        //---------------------------------------------------------------------
        //<Snippet1>
        // Ensure that the schema is in the library and registered with the document.
        private bool CheckSchema()
        {
            const string namespaceUri = "http://schemas.contoso.com/projects";
            bool namespaceFound = false;
            bool namespaceRegistered = false;

            foreach (Word.XMLNamespace n in Application.XMLNamespaces)
            {
                if (n.URI == namespaceUri)
                {
                    namespaceFound = true;
                }
            }

            if (!namespaceFound)
            {
                MessageBox.Show("XML Schema is not in library.");
                return false;
            }

            foreach (Word.XMLSchemaReference r in this.XMLSchemaReferences) 
            {
                if (r.NamespaceURI == namespaceUri)
                {
                    namespaceRegistered = true;
                }
            }

            if (!namespaceRegistered)
            {
                MessageBox.Show("XML Schema is not registered for this document.");
                return false;
            }
            
            return true;
        }
        //</Snippet1>
    }
}
